schema.graphql

```diff
@@ -1,68 +1,83 @@
-schema {
-  query: Query
-  mutation: Mutation
-}
-
 type Game {
-  cluster: String
-  createdAt: AWSDateTime
-  description: String
-  domain: String
-  grade: String
-  id: ID!
-  standard: String
-  title: String!
-  updatedAt: AWSDateTime
+	id: ID!
+	title: String!
+	description: String
+	cluster: String
+	domain: String
+	grade: String
+	standard: String
+	updatedAt: AWSDateTime
+	createdAt: AWSDateTime
+	questions: [Question]
 }
 
 type GameSession {
-  gameCode: Int!
-  gameId: ID!
-  gameSessionID: ID!
-  hintsRevealTime: Int
-  multiDeviceMode: Boolean!
-  startedDateTimeUTC: AWSDateTime!
-  state: GameSessionState!
-  teamIds: [Int]!
-  votingTime: Int
+	gameSessionID: ID!
+	gameId: ID!
+	teamIds: [Int]!
+	startedDateTimeUTC: AWSDateTime
+	gameCode: Int!
+	multiDeviceMode: Boolean!
+	votingTime: Int
+	hintsRevealTime: Int
+	state: GameSessionState!
+	random: AWSJSON
+}
+
+type GameSessionConfig {
+	configVersion: Int!
+	pointsForMostPopularTrickAnswer: Int!
+	pointsPerCorrectAnswer: Int!
+	votingTime: Int!
+	hintsRevealTime: Int!
+}
+
+input GameSessionInput {
+	gameId: ID!
+	teamIds: [Int]
+	startedDateTimeUTC: AWSDateTime
+	gameCode: Int
+	multiDeviceMode: Boolean!
+	votingTime: Int
+	hintsRevealTime: Int
+	state: GameSessionState!
+}
+
+enum GameSessionState {
+	NotStarted
+	InitialIntro
+	ChoosingTrickAnswer
+	Voting
+	ReviewingResult
+	Finished
 }
 
 type Mutation {
-  saveGameSession(gameSession: GameSessionInput): GameSession
+	saveGameSession(gameSession: GameSessionInput): GameSession
+	joinGameSession(gameCode: Int!): GameSession!
+	joinGame(gameCode: Int!): GameSession
+	updateGame(gameId: ID!): Game!
```

Comment:
```suggestion
	updateGame(game: Game!): Game!
```
Because this mutation will edit one or more fields on the game, I think we should pass it an entire Game object.

cc/ @EricNguyen-B @maniramezan